### PR TITLE
Docs to run a spark streaming application with Kafka.

### PR DESCRIPTION
**Pull request by averma-canonical**

### CODE REVIEW

**Comment by deusebio** [comment]
I have tested the process and it worked fine. We need to modify the index.md as well to index the new content. Before merging I will also create the topics in Discourse and make sure the content in the repo is aligned

File: docs/how-to/spark_streaming.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+spark-client.service-account-registry list
+
+spark-client.pyspark --username hello --namespace spark-streaming --conf spark.executor.instances=1 --conf spark.jars.ivy=/tmp --packages org.apache.spark:spark-streaming-kafka-0-10_2.12:3.2.0,org.apache.spark:spark-sql-kafka-0-10_2.12:3.2.0
+```
+
+Within the pyspark shell, now use the credentials retrieved previously to read stream from Kafka.
+
+```python
+from pyspark.sql.functions import explode
+from pyspark.sql.functions import split, col
```
Comment (deusebio) [inline docs/how-to/spark_streaming.md:99]:
```python 
from pyspark.sql.functions import udf
from json import loads
```

File: docs/how-to/spark_streaming.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+          .option("kafka.sasl.mechanism", "SCRAM-SHA-512") \
+          .option("kafka.security.protocol", "SASL_PLAINTEXT") \
+          .option("kafka.sasl.jaas.config", f'org.apache.kafka.common.security.scram.ScramLoginModule required username={username} password={password};') \
+          .option("subscribe", "spark-streaming-store") \
+          .option("includeHeaders", "true") \
+          .load()
+
+words = lines.select(explode(split(col("value"), " ")).alias("word"))
+
+wordCounts = words.groupBy("word").count()
```
Comment (deusebio) [inline docs/how-to/spark_streaming.md:115]:
```python 
get_origin = udf(lambda x: loads(x)["origin"])
count = lines.withColumn("origin", get_origin(col("value"))).select("origin")\
	.groupBy("origin", "partition")\
	.count()
```

File: docs/how-to/spark_streaming.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+query = wordCounts \
```
Comment (deusebio) [inline docs/how-to/spark_streaming.md:117]:
```
query = count 
```

**Review by deusebio — COMMENTED** [review]
I have just slightly revised the pyspark example to have some more meaningful information out. The word count was not too relevant in this case. 

Now I'm doing a count by partition and origin

**Review by welpaolo — APPROVED** [review]
LGTM

File: docs/how-to/spark_streaming.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+          .option("subscribe", "spark-streaming-store") \
+          .option("includeHeaders", "true") \
+          .load()
+
+get_origin = udf(lambda x: loads(x)["origin"])
+count = lines.withColumn("origin", get_origin(col("value"))).select("origin")\
+          .groupBy("origin", "partition")\
+          .count()
+
+query = count
```
Comment (deusebio) [inline docs/how-to/spark_streaming.md:118]: I meant...just put

```
count.awaitTermination()
```

File: docs/how-to/spark_streaming.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+query.awaitTermination()
```
Comment (deusebio) [inline docs/how-to/spark_streaming.md:120]:
```suggestion
count.awaitTermination()
```

File: docs/how-to/spark_streaming.md (hunk 4 above)
Comment (deusebio) [inline docs/how-to/spark_streaming.md:118]:
```suggestion
```

**Review by deusebio — APPROVED** [review]
Just small improvements, but feel free to merge! Looks good!

**Review by deusebio — APPROVED** [review]
LGTM!

**Pull request merged**